src/mpid/ch4/netmod/ucx/ucx_probe.h

```diff
@@ -36,18 +36,22 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_NM_mpi_improbe(int source,
         req = (MPIR_Request *) MPIR_Request_create(MPIR_REQUEST_KIND__MPROBE, 0);
         MPIR_ERR_CHKANDSTMT((req) == NULL, mpi_errno, MPIX_ERR_NOREQ, goto fn_fail, "**nomemreq");
         MPIR_Request_add_ref(req);
-        MPIDI_UCX_REQ(req).message_handler = message_h;
+
+        MPIDI_UCX_req_mprobe_t *mprobe = MPIDI_UCX_REQ(req);
+        mprobe->message_h = message_h;
 
         if (status != MPI_STATUS_IGNORE) {
             status->MPI_SOURCE = MPIDI_UCX_get_source(info.sender_tag);
             status->MPI_TAG = MPIDI_UCX_get_tag(info.sender_tag);
             count = info.length;
             MPIR_STATUS_SET_COUNT(*status, count);
         }
+
+        *message = req;
     } else {
         *flag = 0;
+        *message = NULL;
```

Comment:
We could. It is a bit defensive programming. I find the code cleaner if we spell out all the paths. In addition, setting `*message` to `NULL` here so it can trigger better errors if upper layer accidentally miss-use it.